libs/payments/cart/src/lib/cart.service.ts

```diff
@@ -449,41 +456,52 @@ export class CartService {
       couponCode: cart.couponCode || undefined,
     });
 
-    // Cart latest invoice data
-    let latestInvoicePreview: InvoicePreview | undefined;
     let paymentInfo: PaymentInfo | undefined;
-    if (customer && cart.stripeSubscriptionId) {
-      // fetch latest payment info from subscription
-      const subscription = await this.subscriptionManager.retrieve(
-        cart.stripeSubscriptionId
+    if (customer?.invoice_settings.default_payment_method) {
+      const paymentMethodPromise = this.paymentMethodManager.retrieve(
+        customer.invoice_settings.default_payment_method
       );
-      assert(subscription.latest_invoice, 'Subscription not found');
-      latestInvoicePreview = await this.invoiceManager.preview(
-        subscription.latest_invoice
-      );
-
+      const customerSessionPromise = cart.stripeCustomerId
+        ? this.customerSessionManager.create(cart.stripeCustomerId)
+        : undefined;
+      const [paymentMethod, customerSession] = await Promise.all([
+        paymentMethodPromise,
+        customerSessionPromise,
+      ]);
+      paymentInfo = {
+        type: paymentMethod.type,
+        last4: paymentMethod.card?.last4,
+        brand: paymentMethod.card?.brand,
+        customerSessionClientSecret: customerSession?.client_secret,
+      };
+    } else if (subscriptions.length) {
+      const subscription = subscriptions[0];
       // fetch payment method info
       if (subscription.collection_method === 'send_invoice') {
         // PayPal payment method collection
-        // TODO: render paypal payment info in the UI (FXA-10608)
         paymentInfo = {
           type: 'external_paypal',
         };
-      } else {
-        // Stripe payment method collection
-        if (customer.invoice_settings.default_payment_method) {
-          const paymentMethod = await this.paymentMethodManager.retrieve(
-            customer.invoice_settings.default_payment_method
-          );
-          paymentInfo = {
-            type: paymentMethod.type,
-            last4: paymentMethod.card?.last4,
-            brand: paymentMethod.card?.brand,
-          };
-        }
       }
     }
 
+    // Cart latest invoice data
+    let latestInvoicePreview: InvoicePreview | undefined;
+    if (customer && cart.stripeSubscriptionId) {
+      assert(
+        subscriptions.length,
+        'Customer should have at least one subscription'
+      );
```

Comment:
[non-blocking] Can't this assertion be eliminated given the assertion on line 499 just below this?

Reply:
Ah yes, thank you!